pilot/pkg/features/pilot.go

```diff
@@ -447,6 +447,9 @@ var (
 	EnableEnvoyFilterMetrics = env.RegisterBoolVar("PILOT_ENVOY_FILTER_STATS", false,
 		"If true, Pilot will collect metrics for envoy filter operations.").Get()
 
+	EnableRouteCollapse = env.RegisterBoolVar("PILOT_ENABLE_ROUTE_COLLAPSE_OPTIMIZATION", true,
```

Comment:
When do we not this optimization? Is it more to cover for any unforeseen bugs?

Reply:
Basically two issues:
* Maybe be unforeseen bugs
* There is some strange behavior with this that we have to pick one "name" for the VirtualHost, so envoyfilter matching may be impacted

Reply:
Second part is clear to me. May be we should add this to release notes?

Reply:
added